### PR TITLE
feat(mv3): hardware wallet support for trezor

### LINKED ISSUE #21623: Connect Trezor to Offscreen Document API
### What is this about?

Hardware wallets will need to communicate with the extension via an offscreen document (ref: #21622 ) in order to properly work. This work has largely been done in #17687 but needs to be integrated in a way that also works with snaps offscreen communication. Once #21622 lands and #21145 lands we can pull the trezor iframe forward and include it in the offscreen html file.

Connecting to the offscreen document with the new trezor keyring should require only the code from the existing PR and any updates in order to fix bugs with it functioning. This work should be done for each hardware wallet keyring type that requires it one at a time in order to minimize code review and QA requirements. 

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- [ ] A trezor-iframe.html file is included in the offscreen folder and is properly copied to the dist folder during build
- [ ] script to connect the trezor to the iframe for communication and forward to the extension is written, bundled and included in the trezor-iframe.html file with a script tag. 
- [ ]  trezor-iframe is added to the offscreen.html file as the src attribute for an iframe
- [ ] User can connect to and use a trezor with MV3 turned on.

### References

_No response_

### PR DESCRIPTION
**Pull request by brad-decker**
## **Description**

Uses the Offscreen document to add an iframe for Trezor hardware wallet communication. The entry point of the trezor-iframe is trezor-iframe.ts and it is protected by Lavamoat using the build system in the same way that the offscreen document is. 

## **Related issues**

Fixes: #21623 

## **Manual testing steps**

1. run `yarn start:mv3`
2. load the extension into chrome, or reload it if already added.
3. If you previously had your Trezor added to the extension, remove it (or you can add a secondary account from it)
4. Click Add new hardware wallet from the add account dropdown menu. 
5. See that Trezor and QR are valid options
6. Make sure you have Trezor Suite open and your device unlocked. 
7. Click trezor and then connect.
8. See that you are prompted to connect your trezor and select an account to import. 
9. Perform transactions, signatures, etc using the test dapp. 

### **Before**
Screenshot 2023-11-30 at 12 04 49 PM

### **After**
Screenshot 2023-11-30 at 11 59 01 AM
Screenshot 2023-11-30 at 12 01 35 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/lib/offscreen-bridge/trezor-offscreen-bridge.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+          target: OffscreenCommunicationTarget.trezorOffscreen,
+          action: TrezorAction.signTypedData,
+          params,
+        },
+        (response) => {
+          resolve(
+            response as unknown as
+              | Unsuccessful
+              | Success<PROTO.EthereumTypedDataSignature>,
+          );
```
Comment (brad-decker) [inline app/scripts/lib/offscreen-bridge/trezor-offscreen-bridge.ts:137]: Now that I added types in global.d.ts i can get rid of this casting here I believe.

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Review by pedronfigueiredo — DISMISSED** [review]
I manually tested this feature with a Trezor and can confirm it works! LGTM

**Comment by brad-decker** [comment]
> I tested this out and:
> 
> 1. I was able to connect my Trezor -- YAY!
> 2. I tried to complete a _Send_ transaction but the send would repeatedly fail with a `Forbidden key path` error.  The Trezor window _would_ pop up asking me to approve but returning back to MetaMask insta-fails the transaction.

@darkwing did you use eth or a token and what network?

**Comment by brad-decker** [comment]
@darkwing go into trezor suite, go to settings, device, then edit on safety checks and select prompt. Its automatically closing the request becuase of a safety check -- going to confirm this happens on non mv3 to make sure its not a new thing

**Review by legobeat — DISMISSED** [review]

**Comment by darkwing** [comment]
Yeah, getting the same thing.  I was trying to send ETH on Goerli.

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Review by darkwing — APPROVED** [review]

**Comment by brad-decker** [comment]
@SocketSecurity ignore @trezor/transport@1.1.12

**Comment by brad-decker** [comment]
@SocketSecurity ignore @trezor/blockchain-link-types@1.0.2

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**